### PR TITLE
Updated to the latest versions of the protocol tests.

### PR DESCRIPTION
**Pull request by trevorrowe**
Bulk updated the protocol tests. There were some minor library updates necessary to support these.

### CODE REVIEW

File: aws-sdk-core/lib/aws-sdk-core/rest/request/endpoint.rb [hunk 1]
```diff
@@ -73,6 +73,14 @@ def apply_querystring_params(uri, params)
                   raise NotImplementedError, msg
                 end
 
+              when ListShape
+                if StringShape === member.shape.member.shape
+                  parts += list_of_strings(member.location_name, params[member_name])
+                else
+                  msg = "Only list of strings supported"
```
Comment (awood45) [inline aws-sdk-core/lib/aws-sdk-core/rest/request/endpoint.rb]: Optional: Could we output the actual type we received in this case?
```ruby
msg = "Only list of strings supported, was: #{member.shape.member.shape}"
```

**Comment by awood45** [comment]
LGTM

**Comment by trevorrowe** [comment]
I've added more information about the given shape in the unsupported case.

**Comment by trevorrowe** [comment]
I merged this locally, it has been pushed to master already.

**Pull request closed without merging**